src/bin/cargo-flamegraph.rs

```diff
@@ -310,29 +310,45 @@ fn find_unique_target(
         });
     }
 
+    let mut num_packages = 0;
+    let mut is_default = false;
+
     let mut targets: Vec<_> = packages
         .flat_map(|p| {
-            let Package { targets, name, .. } = p;
+            let Package {
+                targets,
+                name,
+                default_run,
+                ..
+            } = p;
+            num_packages += 1;
+            if default_run.is_some() {
+                is_default = true;
+            }
             targets.into_iter().filter_map(move |t| {
-                t.kind
-                    .iter()
-                    .any(|s| kind.contains(&s.as_str()))
-                    .then(|| BinaryTarget {
-                        package: name.clone(),
-                        target: t.name,
-                        kind: t.kind,
-                    })
+                let ok_kind = t.kind.iter().any(|s| kind.contains(&s.as_str()));
+                let default_filter = default_run
+                    .as_ref()
+                    .map_or(true, |default_name| &t.name == default_name);
```

Comment:
I'd like this rewritten as a `match` expression. Also probably good to add some comments inside the `filter_map()` closure here about what it is we're trying to achieve.